### DISCUSSION TITLE
libsslocal exits too fast (exit code: 134)            device(amazon firetv stick 4k)

### DISCUSSION THREAD

**Discussion by scola** [shadowsocks/shadowsocks-android#3184] [Q&A]
Please read contributing guidelines. Thanks.

**Describe the bug**
when start service, popup toast: libsslocal exits too fast (exit code: 134)

**To Reproduce**
1. start ss service (press launch button)
2. shadowsocks version: 5.2.5 (universal)

**Expected behavior**
button turn green, vpn service started

**Screenshots**

**Smartphone (please complete the following information):**
 -  Amazon firetv stick 4k

**Configuration**
Put an `x` inside the [ ] that applies.

* [ ] IPv4 server address
* [ ] IPv6 server address
* [ ] Client IPv4 availability
* [ ] Client IPv6 availability
* Encrypt method:
* Route
  * [ ] All
  * [ ] Bypass LAN
  * [ ] Bypass China
  * [ ] Bypass LAN & China
  * [ ] GFW List
  * [ ] China List
  * [ ] Custom rules
* [ ] IPv6 route
* [ ] Apps VPN mode
  * [ ] Bypass mode
* Remote DNS: 8.8.8.8
* [ ] DNS over UDP
* Plugin configuration (if applicable):
* [ ] Auto Connect
* [ ] TCP Fast Open
* If you're not using VPN mode, please supply more details here:

**Additional context**

mantis:/ $ cat /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 11.60
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 1
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 11.60
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 2
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 11.60
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 3
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 11.60
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

Hardware        : Generic DT based system
Revision        : 0000
Serial          : e2e411d2d8e88eca
mantis:/ $

logcat : 
[shadowsocks.log](https://github.com/shadowsocks/shadowsocks-android/files/6989972/shadowsocks.log)

**Comment by scola** [comment]
08-16 13:43:06.006  8004  8004 F DEBUG   : 
08-16 13:43:06.006  8004  8004 F DEBUG   : backtrace:
08-16 13:43:06.006  8004  8004 F DEBUG   :     #00 pc 00049ca4  /system/lib/libc.so (tgkill+12)
08-16 13:43:06.006  8004  8004 F DEBUG   :     #01 pc 00047443  /system/lib/libc.so (pthread_kill+34)
08-16 13:43:06.006  8004  8004 F DEBUG   :     #02 pc 0001d705  /system/lib/libc.so (raise+10)
08-16 13:43:06.006  8004  8004 F DEBUG   :     #03 pc 00019251  /system/lib/libc.so (__libc_android_abort+34)
08-16 13:43:06.006  8004  8004 F DEBUG   :     #04 pc 00017210  /system/lib/libc.so (abort+4)
08-16 13:43:06.006  8004  8004 F DEBUG   :     #05 pc 0012e983  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.006  8004  8004 F DEBUG   :     #06 pc 0012e8df  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #07 pc 001bae37  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #08 pc 001bacb9  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #09 pc 001cb55d  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #10 pc 001cb4fb  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #11 pc 001cb4c3  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #12 pc 000313bf  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #13 pc 0008ecf5  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #14 pc 000a9a3f  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #15 pc 00049999  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #16 pc 000ad063  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so
08-16 13:43:06.007  8004  8004 F DEBUG   :     #17 pc 00016e29  /system/lib/libc.so (__libc_init+48)
08-16 13:43:06.007  8004  8004 F DEBUG   :     #18 pc 00035cc0  /data/app/com.github.shadowsocks-2/lib/arm/libsslocal.so

**Comment by scola** [comment]
adb shell getprop ro.build.version.release
7.1.2

**Comment by Mygod** [comment]
Thanks for the comprehensive report. From log: `thread 'main' panicked at 'aborted with Address already in use (os error 98)', bin/sslocal.rs:470:45`

Try to use a different port.

**Comment by scola** [comment]
shadowsocks android can not config local_port, can you add this feature?
currently I failed to compile this project

> Task :core:cargoBuildArm FAILED
   Compiling cfg-if v1.0.0
   Compiling scopeguard v1.1.0
   Compiling matches v0.1.8
   Compiling pin-project-lite v0.2.7
   Compiling smallvec v1.6.1
   Compiling once_cell v1.8.0
   Compiling tinyvec_macros v0.1.0
   Compiling ppv-lite86 v0.2.10
   Compiling futures-sink v0.3.16
   Compiling lazy_static v1.4.0
   Compiling percent-encoding v2.1.0
   Compiling slab v0.4.3
   Compiling futures-io v0.3.16
   Compiling pin-utils v0.1.0
   Compiling linked-hash-map v0.5.4
   Compiling bytes v1.0.1
error[E0463]: can't find crate for `core`

**Comment by scola** [comment]
wifi      2813  1     4004   868            0 00000000 S /system/bin/socks
this process use port 1080 and I do not have permission to kill it

**Comment by Mygod** [comment]
Check settings - advanced.

**Comment by scola** [comment]
@Mygod really thanks for your guide
My problem has been resolved. maybe two things need to improve
1. when local_port 1080 is used by other process. user need more reasonable tips, not crash directly
2. Setting page can not focus to select advance by remote controller (I resolved it by [scrcpy](https://github.com/Genymobile/scrcpy)), I did not use shadowsocks tv version, maybe this issue is not important
